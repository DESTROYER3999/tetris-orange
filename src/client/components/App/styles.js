import styled from "styled-components";

export const StyledContainer = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  &::after {
    content: "";
    background: url(assets/tetris-theme.jpg);
    background-size: cover;
    opacity: 0.2;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    position: absolute;
    z-index: -1;
  }

  // background-size: cover;
`;
