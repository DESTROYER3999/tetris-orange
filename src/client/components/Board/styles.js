import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  position: relative;
  align-self: center;
  background-color: rgba(0, 0, 0, 0.3);
  display: flex;
  flex-direction: column;
  height: 86vh;
  width: 43vh;
  border: 3px solid white;
  border-radius: 3px 0px 0px 3px;

  ${props =>
    props.isSpectre &&
    css`
      border: 2px solid white;
      border-radius: 2px;
      height: 30vh;
      width: 15vh;
      background-color: rgba(0, 0, 0, 0.3);
    `};
`;

export const StyledRow = styled.div`
  flex: 1;
  display: flex;
`;
