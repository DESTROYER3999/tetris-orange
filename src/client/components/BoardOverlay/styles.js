import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  align-self: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  border: 3px solid white;
  border-radius: 5px;
  background-color: rgba(0, 0, 0, 0.5);
  position: absolute;
  top: -3px;
  height: 86vh;
  width: 43vh;
  h2 {
    margin-top: 40px;
    align-self: center;
  }
`;

export const StyledGameOverText = styled.h1`
  align-self: center;
  font-size: 5vh;
`;
