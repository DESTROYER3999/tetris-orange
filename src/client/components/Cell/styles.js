import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  flex: 1 10%;
  border: 1px solid black;
  ${props =>
    props.color &&
    !props.isSpectre &&
    css`
      border: 5px outset ${props => props.color};
      background-color: ${props => props.color};
    `};
  ${props =>
    props.isSpectre &&
    css`
      border: 0px;
      ${props =>
        props.color &&
        css`
          background: #ff8a0080;
        `};
    `};
`;

export const StyledLetter = styled.p`
  font-size: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: font-size 0.2;
  &:hover {
    font-size: 30px;
  }
`;
