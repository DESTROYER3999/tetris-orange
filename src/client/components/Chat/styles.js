import styled from "styled-components";

export const StyledMainContainer = styled.div`
  display: flex;
  flex: 4;
  padding: 10px;
  margin: 5px;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 2px;
`;

export const StyledChatContainer = styled.div`
  display: flex;
  flex: 1;
  border: 3px solid white;
  border-radius: 5px;
  height: 0;
  min-height: 100%;
`;

export const StyledLeftContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  background-color: rgba(0, 0, 0, 0.2);
  border-right: 3px solid white;
`;

export const StyledRightContainer = styled.div`
  flex: 4;
  display: flex;
  flex-direction: column;
`;

export const StyledTitleContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 4vh;
  border-bottom: 3px solid white;
`;
export const StyledMessagesContainer = styled.div`
  display: block;
  flex: 1;
  padding: 5px;
  flex-wrap: wrap;
  overflow-y: auto;
  overflow-x: hidden;
  text-overflow: ellipsis;
`;

export const StyledForm = styled.form`
  display: flex;
  padding: 5px;
`;

export const StyledPlayerListContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  align-items: center;
  padding: 5px;
  border-bottom: 5px;
  overflow-y: auto;
  overflow-x: hidden;
  text-overflow: ellipsis;
`;

export const StyledText = styled.div`
  font-family: Verdana;
  font-size: 1.5vh;
  min-width: 100%;
  width: 0;
  & font {
    font-family: Verdana;
    font-weight: 600;
  }
`;

export const StyledInput = styled.input`
  height: 4vh;
  padding: 0.5vh;
  padding-left: 1vh;
  border-radius: 5px;
  border: 2px solid white;
  font-size: 2vh;
  width: 100%;
  background: none;
  outline: none;
  color: white;
  font-family: Verdana;
  font-size: 1.5vh;
  ::placeholder {
    color: white;
  }
`;
