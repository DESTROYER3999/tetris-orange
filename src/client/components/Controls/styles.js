import styled from "styled-components";

export const StyledContainer = styled.div`
  align-self: center;
  display: flex;
  flex-direction: column;
  width: 70%;
  div {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    kbd {
      font-size: 2vh;
      border-radius: 3px;
      padding: 1px 2px 0;
      border: 3px solid white;
    }
    span {
      font-size: 1.5vh;
    }
  }
`;
