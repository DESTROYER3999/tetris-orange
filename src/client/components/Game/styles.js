import styled from "styled-components";

export const StyledContainer = styled.div`
  padding: 1vh;
  flex: 1;
  display: flex;
`;

export const StyledColumnContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex-basis: 20%;
  height: 86vh;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 0px 3px 3px 0px;
  border: 3px solid white;
  align-items: center;
  .value {
    font-size: 2vh;
  }
`;

export const StyledLeftItem = styled.div`
  flex: 2;
  padding: 1vh;
  margin: 5px;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 2px 2px 2px 60px;
  display: flex;
  justify-content: space-around;
  align-items: center;
`;

export const StyledRightItem = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
`;

export const StyledLabel = styled.div`
  font-size: 3vh;
  padding: 1vh;
`;
