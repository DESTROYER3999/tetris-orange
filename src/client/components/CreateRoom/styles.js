import styled from "styled-components";
import Input from "../common/Input";
import { withStyles } from "@material-ui/core/styles";

export const StyledForm = styled.form`
  align-self: center;
  width: 70%;
  display: flex;
  padding: 1vh;
  border-radius: 4px;
  justify-content: center;
  align-items: center;
  background: rgb(0, 0, 0, 0.2);
`;

export const StyledRoomNameContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const StyledLabel = styled.label`
  padding: 0.5vh;
  font-size: 2vh;
  color: white;
`;

export const StyledButton = styled.button`
  height: 2vh;
  background: none;
  border: 0px;
  outline: none;
  opacity: 0;
  &:enabled {
    opacity: 1;
  }
  &:disabled {
    heigth: 0px;
  }
`;

export const StyledInput = withStyles({
  root: {
    "&": {
      marginTop: 0,
      marginRight: 10
    },
    "& label.Mui-focused": {
      color: "white"
    },
    "$ .MuiFormHelperText-contained p": {
      height: "2vh"
    },
    "& label": {
      color: "white"
    },
    "& input": {
      color: "white"
    },
    "& .MuiFormHelperText-root": {
      backgroundColor: "white",
      fontSize: "1.5vh"
    },
    "& .MuiOutlinedInput-root": {
      "& fieldset": {
        borderColor: "white"
      },
      "&:hover fieldset": {
        borderColor: "white"
      },
      "&.Mui-focused fieldset": {
        borderColor: "white"
      },
      "&.Mui-error fieldset": {
        borderColor: "white"
      }
    }
  }
})(Input);
