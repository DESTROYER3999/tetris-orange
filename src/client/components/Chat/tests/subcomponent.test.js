import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ChatSub from "../subcomponent";
import socket from "../../../services/socket-api";
import { StyledInput, StyledForm } from "../styles";

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

const props = {
  chatMessages: [
    { type: "message", author: "Player1", text: "Hello world" },
    { type: "notification", text: "Player1 joined Lobby" }
  ],
  players: ["Player1", "Player2"],
  onSubmit: jest.fn()
};

describe("<ChatSub />", () => {
  test("ChatSub rendering without crashing", () => {
    const wrapper = shallow(<ChatSub {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("ChatSub input remove kaydown listener on focus event", () => {
    document.removeEventListener = jest.fn();

    const wrapper = shallow(<ChatSub {...props} />);
    wrapper.find(StyledInput).simulate("focus");
    expect(document.removeEventListener).toHaveBeenCalled();
  });
  test("ChatSub input add keydown listener on blur", () => {
    document.addEventListener = jest.fn();

    const wrapper = shallow(<ChatSub {...props} />);
    wrapper.find(StyledInput).simulate("blur");
    expect(document.addEventListener).toHaveBeenCalled();
  });
  test("ChatSub input add keydown listener on blur event", () => {
    document.addEventListener = jest.fn();

    const wrapper = shallow(<ChatSub {...props} />);
    wrapper.find(StyledInput).simulate("blur");
    expect(document.addEventListener).toHaveBeenCalled();
  });
  test("ChatSub input value change on change event // input value reset and onSubmit called on submit event", () => {
    const wrapper = shallow(<ChatSub {...props} />);
    const message = "Hello world !";
    const event = { target: { value: message } };

    wrapper.find(StyledInput).simulate("change", event);
    expect(wrapper.find(StyledInput).props().value).toBe(message);
    wrapper.find(StyledForm).simulate("submit", {
      preventDefault: () => {}
    });
    expect(wrapper.find(StyledInput).props().value).toBe("");
    expect(props.onSubmit).toHaveBeenCalled();
  });
});
