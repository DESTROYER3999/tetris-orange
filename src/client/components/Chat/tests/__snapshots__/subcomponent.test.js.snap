// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatSub /> ChatSub rendering without crashing 1`] = `
<styled.div
  id="chat"
>
  <styled.div>
    <styled.div>
      <styled.div>
        <p>
          Players
        </p>
      </styled.div>
      <styled.div>
        <styled.div
          key="0"
        >
          Player1
        </styled.div>
        <styled.div
          key="1"
        >
          Player2
        </styled.div>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <p>
          Messages
        </p>
      </styled.div>
      <styled.div>
        <styled.div
          key="0"
        >
          <font>
            Player1
            :
          </font>
           
          Hello world
        </styled.div>
        <styled.div
          key="1"
        >
          <font
            style={
              Object {
                "color": undefined,
              }
            }
          >
            Player1 joined Lobby
          </font>
        </styled.div>
        <div />
      </styled.div>
      <styled.form
        onSubmit={[Function]}
      >
        <styled.input
          autoComplete="off"
          id="chatMessage"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Write a message ..."
          spellCheck="false"
          type="text"
          value=""
        />
      </styled.form>
    </styled.div>
  </styled.div>
</styled.div>
`;
